static/js/modules/reaction-box.js

```diff
@@ -3,36 +3,95 @@
 /* global module, ga */
 
 var $ = require('jquery');
+var helpers = require('./helpers');
 
 function ReactionBox(selector) {
   this.$element = $(selector);
-  this.$form = $(selector).find('form');
-  this.$fieldset = this.$form.find('fieldset');
-  this.$textarea = this.$form.find('textarea');
-  this.$radios = this.$form.find('ul');
-  this.boxID = this.$element.data('box-id');
+  this.$form = this.$element.find('form');
+  this.$fieldset = this.$element.find('fieldset');
+  this.$textarea = this.$element.find('textarea');
+
+  this.$step1 = this.$element.find('.js-reaction-step-1');
+  this.$step2 = this.$element.find('.js-reaction-step-2');
+  this.$success = this.$element.find('.js-reaction-success');
+  this.$error = this.$element.find('.js-reaction-error');
+
+  this.name = this.$element.data('name');
+  this.location = this.$element.data('location');
+
+  this.url = helpers.buildAppUrl(['issue']);
+
   this.$fieldset.on('change', this.handleChange.bind(this));
+  this.$element.on('click', '.js-skip', this.handleSuccess.bind(this));
+  this.$element.on('click', '.js-reset', this.handleReset.bind(this));
+
+  this.$form.on('submit', this.handleSubmit.bind(this));
 }
 
 ReactionBox.prototype.handleChange = function(e) {
   var $target = $(e.target);
+  this.reaction = $target.val();
 
-  if (typeof ga === 'undefined') { return; }
-  var val = $target.val();
-  var gaEventData = {
-    hitType: 'event',
-    eventCategory: 'Reactions: ' + this.boxID,
-    eventAction: val
-  };
-
-  ga('send', gaEventData);
+  if (typeof ga !== 'undefined') {
+    var gaEventData = {
+      hitType: 'event',
+      eventCategory: 'Reactions',
+      eventAction: this.location + '-' + this.name + ': ' + this.reaction,
+      eventValue: 1
+    };
+    ga('send', gaEventData);
+  }
```

Comment:
This should really be a helper method that would detect `ga` and do a no-op if it's unavailable (or even queue it if there's a possibility ga will load later).